src/client/proxy/http.cc

```diff
@@ -500,15 +503,24 @@ bool HTTPMessage::SaveJumpServiceAddress()
 {
   try
     {
+      if (m_Base64Destination.empty())
+        {
+          LOG(debug) << "HTTPProxyHandler: empty Base64 destination";
+          return false;
+        }
+      if (m_Address.empty())
+        ParseURL();
+      core::IdentityEx ident;
+      // Extra sanity check to ensure base64 destination is uri-decoded
+      ident.FromBase64(boost::network::uri::decoded(m_Base64Destination));
```

Comment:
By now, you shouldn't need to parse anything more than once.

Reply:
Right, I was considering not adding this extra URI-decoding step. Reverted, and will keep in mind for the HTTP proxy rewrite.